### PR TITLE
fix(bi-link): isRelativePath invalid in vuepress

### PR DESCRIPTION
**Pull request by LincZero**

### CODE REVIEW

File: packages/markdown-it-bi-directional-links/src/index.ts [hunk 1]
```diff
@@ -355,9 +355,20 @@ export const BiDirectionalLinks: (options?: BiDirectionalLinksOptions) => Plugin
 
       let resolvedNewHref: string
       if (isRelativePath) {
-        resolvedNewHref = relative(dirname(state.env.relativePath), matchedHref)
-          .split(sep)
-          .join('/')
+        if (state.env.relativePath) { // linux
```
Comment (nekomeowww) [inline packages/markdown-it-bi-directional-links/src/index.ts:358]: Hmmm. 

This behavior is weird. We should write the comments to tell the later coming developers why these properties, and perhaps we can file up an issue to VitePress to ask them whether this is the unexpected behavior.

Comment (nekomeowww) [reply packages/markdown-it-bi-directional-links/src/index.ts:358]: Can we some how write some tests to test these two branches and cobfirm that they are working correctly?

I can help you to configure the test CI to run on both Unix-like OS and Windows.

Comment (LincZero) [reply packages/markdown-it-bi-directional-links/src/index.ts:358]: Some notes: https://github.com/nolebase/integrations/pull/344

Comment (LincZero) [reply packages/markdown-it-bi-directional-links/src/index.ts:358]: 391230602-807ff1d5-184d-4347-8242-53383e024349-1

Left: Windows、Right: Linux

Comment (LincZero) [reply packages/markdown-it-bi-directional-links/src/index.ts:358]: 这个是Markdown-it的state，大概定位了一下：

https://github.com/markdown-it/markdown-it/blob/0fe7ccb4b7f30236fb05f623be6924961d296d3d/lib/rules_core/state_core.mjs#L8

image

不看代码实现的话没啥头绪，但不想看。我写代码倒没这么严谨。反正这次PR至少不会更糟糕，就不管了。多平台CI倒是可以做

**Comment by nekomeowww** [comment]
Rebase / Merge is required.

**Comment by nekomeowww** [comment]
As suggested in https://github.com/nolebase/integrations/pull/361#discussion_r1872462228

> We should write the comments to tell the later coming developers why these properties, and perhaps we can file up an issue to VitePress to ask them whether this is the unexpected behavior.

Or at least, link this PR in the comment.

File: packages/markdown-it-bi-directional-links/src/index.ts [hunk 2]
```diff
@@ -355,9 +355,20 @@ export const BiDirectionalLinks: (options?: BiDirectionalLinksOptions) => Plugin
...
       if (isRelativePath) {
-        resolvedNewHref = relative(dirname(state.env.relativePath), matchedHref)
-          .split(sep)
-          .join('/')
+        if (state.env.relativePath) { // VitePress
+          resolvedNewHref = relative(dirname(state.env.relativePath), matchedHref)
+            .split(sep)
+            .join('/')
+        }
+        else if (state.env.filePathRelative) { // VuePress, see [#361](nolebase/integrations/pull/361) for details
```
Comment (nekomeowww) [inline packages/markdown-it-bi-directional-links/src/index.ts:363]:
```suggestion
        else if (state.env.filePathRelative) { // VuePress, see https://github.com/nolebase/integrations/pull/361 for details
```

**Review by nekomeowww — CHANGES_REQUESTED** [review]

**Review by nekomeowww — APPROVED** [review]

**Pull request merged**